python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -118,10 +118,20 @@ class Value extends TObject {
         )
     }
 
-    /** Gets the boolean value of this value. */
-    boolean booleanValue() {
+    /** Gets the boolean interpretation of this value.
+      * Could be both `true` and `false`, if we can't determine the result more precisely.
+      */
+    boolean getABooleanValue() {
         result = this.(ObjectInternal).booleanValue()
     }
+
+    /** Gets the boolean interpretation of this value, only if we can determine the result preciely.
+      * The result can be `none()`, but never both `true` and `false`.
+      */
+    boolean getDefiniteBooleanValue() {
```

Comment:
As we discussed in person, this would probably make sense to generalise to all values. I'm thinking something along the lines of `ControlFlowNode::hasUniquePointsTo()` (likewise for `Expr`).

Reply:
as we just discussed, the problem wasn't that the ControlFlowNode pointed to multiple Values, but that a single Value could be either true or false. So we decided to keep it like it is in the PR